oktoberfest/data/spectra.py

```diff
@@ -291,14 +297,46 @@ def convert_to_df(self) -> pd.DataFrame:
 
         if "mz" in list(self.layers):
             mz_cols = pd.DataFrame(self.get_matrix(FragmentType.MZ)[0].toarray())
-            mz_cols.columns = self._gen_column_names(FragmentType.MZ)
+            mz_cols.columns = self._gen_column_names(FragmentType.MZ)  # , set(self.obs["FRAGMENTATION"]))
             df_merged = pd.concat([df_merged, mz_cols], axis=1)
         if "raw_int" in list(self.layers):
             raw_cols = pd.DataFrame(self.get_matrix(FragmentType.RAW)[0].toarray())
-            raw_cols.columns = self._gen_column_names(FragmentType.RAW)
+            raw_cols.columns = self._gen_column_names(FragmentType.RAW)  # , set(self.obs["FRAGMENTATION"]))
             df_merged = pd.concat([df_merged, raw_cols], axis=1)
         if "pred_int" in list(self.layers):
             pred_cols = pd.DataFrame(self.get_matrix(FragmentType.PRED)[0].toarray())
-            pred_cols.columns = self._gen_column_names(FragmentType.PRED)
+            pred_cols.columns = self._gen_column_names(FragmentType.PRED)  # , set(self.obs["FRAGMENTATION"]))
             df_merged = pd.concat([df_merged, pred_cols], axis=1)
         return df_merged
+
+    def assemble_df_for_parquet(self, include_intensities: bool = False) -> pd.DataFrame:
+        """
+        Returns a Pandas dataframe that can be serialized to Parquet for building a DLomix dataset.
+
+        :param include_intensities: Whether to include raw intensity values (i.e. labels required for training a model,
+            but not for inference)
+
+        :return: Pandas DataFrame with column names matching to those required for DLomix datasets
+        """
+        frag_dict = {
+            "CID": 1,
+            "HCD": 2,
+            "electron transfer dissociation": 3,
+            "ETD": 3,
+        }  # TODO get frag dict from constants in spectrum fundamentals
```

Comment:
yes, and once you do that please add ETHCD, ETCID, UVPD, EID, ECD, AIECD just so that we have it already in there